Feat/posts only editable for ten minutes

Hives opened the pull request.

Hives committed:
Create 'isEditable' method on Post model

Returns true if post is less than 10 minutes old, false otherwise

Hives committed:
Add timecop gem

For travelling through time

With Jean-Claude Van Damme

Hives committed:
Post#update won't edit a post after 10 minutes

Hives committed:
Make some instance variables local in posts_controller

Keepin that scope TITE

Hives commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Users can only edit their posts up to 10 minutes after they were posted [SMALL]](https://trello.com/c/vqfWQEOm/56-users-can-only-edit-their-posts-up-to-10-minutes-after-they-were-posted-small)

arthurashman commented on Gemfile:
```diff
@@ -38,17 +38,18 @@ group :development, :test do
+  gem "simplecov"
+  gem "simplecov-console"
+  gem "timecop"
 end
```
Can we get these in alphabetical order at some point?

arthurashman commented on app/controllers/posts_controller.rb:
```diff
@@ -35,9 +35,14 @@ def create
+    post = Post.find(params[:id])
+
+    unless post.isEditable
+      flash[:danger] = "Could not edit post. Posts are only editable for 10 minutes."
```
Would it be at all possible to hide/grey out the edit button after 10 minutes?

j-dav replied:
See index.html.erb, we have put this in on lines 23 to 25 already
Code:
        <% if post.isEditable %>
          <p><%= link_to 'Edit', edit_post_path(post) %></p> 
        <% end %>

Hives commented:
If you leave the page open for 10 minutes then the link won't disappear
though which I think is what Arthur is suggesting? But I think that's such
a niche issue it's not worth worrying about, certainly at this stage.

On Mon, 29 Apr 2019, 19:38 James, <notifications@github.com> wrote:

> *@hybridbad* commented on this pull request.
> ------------------------------
>
> In app/controllers/posts_controller.rb
> <https://github.com/Hives/acebook-business-logic/pull/37#discussion_r279487098>
> :
>
> > @@ -35,9 +35,14 @@ def create
>    end
>
>    def update
> -    @post = Post.find(params[:id])
> +    post = Post.find(params[:id])
> +
> +    unless post.isEditable
> +      flash[:danger] = "Could not edit post. Posts are only editable for 10 minutes."
>
> See index.html.erb, we have put this in on lines 23 to 25
> Code:
> <% if post.isEditable %>
>
> <%= link_to 'Edit', edit_post_path(post) %>
>
> <% end %>
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Hives/acebook-business-logic/pull/37#discussion_r279487098>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAGO3S4BY5MOO2YMVSG3653PS46B5ANCNFSM4HJEYWLQ>
> .
>

Hives commented on app/controllers/posts_controller.rb:
```diff
@@ -35,9 +35,14 @@ def create
-    @post = Post.find(params[:id])
+    post = Post.find(params[:id])
+
+    unless post.isEditable
```
I just realised I gave these methods JS-style names. Should be snake_case.

Hives replied:
And Ruby convention for predicate methods is that it should be like `.editable?`

j-dav replied:
Agreed. Go with editable?

Hives replied:
i've updated the PR

Hives commented on app/models/post.rb:
```diff
@@ -3,4 +3,14 @@
+
+  private
+
+  def isLessThan10MinutesOld
```
Should be `.less_than_ten_minutes_old?`

j-dav replied:
Goddam language conventions

Hives commented on app/models/post.rb:
```diff
@@ -3,4 +3,14 @@
   belongs_to :user
   delegate :email, :to => :user
+  
+  def isEditable
```
Should be `.editable?`

j-dav commented:
Oh I see... Yeah probably some jquery implementation or something for that!

Hives committed:
Update Post predicate methods to use Ruby convention

I'd written it as `Post#isEditable` which is JS style. Updated it to
`Post#editable?`

arthurashman commented:
I'm happy with what's here. I don't think there's a need to disappear the links at this point
